python-threatexchange/threatexchange/extensions/video_vpdq/video_vpdq.py

```diff
@@ -0,0 +1,95 @@
+# Copyright (c) Facebook, Inc. and its affiliates. All Rights Reserved
+
+"""
+Wrapper around the vpdq signal type.
+"""
+
+import vpdq  # type: ignore
+from .vpdq_util import json_to_vpdq, vpdq_to_json
+from .vpdq_brute_hasher import match_VPDQ_hash_brute
+import pathlib
+import typing as t
+from threatexchange.content_type.content_base import ContentType
+from threatexchange.content_type.video import VideoContent
+import re
+from threatexchange.signal_type import signal_base
+
+
+class VideoVPDQSignal(signal_base.SimpleSignalType, signal_base.TextHasher):
+    """
+    Simple signal type for video using VPDQ.
+    Read about VPDQ at https://github.com/facebook/ThreatExchange/tree/main/vpdq
+    """
+
+    INDICATOR_TYPE = "HASH_VIDEO_VPDQ"
+    VPDQ_CONFIDENT_DISTANCE_THRESHOLD = 31
+    VPDQ_CONFIDENT_QUALITY_THRESHOLD = 50
+
+    @classmethod
+    def get_content_types(self) -> t.List[t.Type[ContentType]]:
+        return [VideoContent]
+
+    @classmethod
+    def validate_signal_str(cls, signal_str: str) -> str:
+        """VPDQ hex str contains 64 hexidecimal characters."""
+        vpdq_hashes = json_to_vpdq(signal_str)
+        for hash in vpdq_hashes:
+            if not re.match("^[0-9a-f]{64}$", hash.hex):
+                raise ValueError("invalid VPDQ hash")
+            if hash.quality < 0 or hash.quality > 100:
+                raise ValueError("invalid VPDQ hash")
+            if hash.frame_number < 0:
+                raise ValueError("invalid VPDQ hash")
+        return signal_str
+
+    @classmethod
+    def hash_from_file(cls, path: pathlib.Path) -> str:
+        vpdq_hashes = vpdq.computeHash(str(path))
+        return vpdq_to_json(vpdq_hashes)
+
+    @classmethod
+    def hash_from_str(cls, path: str) -> str:
+        vpdq_hashes = vpdq.computeHash(path.rstrip())
+        return vpdq_to_json(vpdq_hashes)
```

Comment:
blocking: This method is supposed to handle when the content type itself is unicode characters (text), which image files are not